### PR TITLE
r_d16: Coffee shop

### PR DESCRIPTION
**Pull request by ShekhaninKyryl**
Please review this PR

```
Створити класс Rectangle з полями float length, height та методами float getArea(), float getPerimeter(); 
a. Реалізувати конструктори: без параметрів(default c-tor), з двома параметрами;

2. Створити клас Coffee з полями: тип напою(enum class), вартість.
В конструктор приходить тип напою, для кожного напою зберігається своя вартість. Реалізувати методи:

void prepare() - надрукувати інформацію про те, який напій готується(складові, час, тд);
int getCost() const - повернути ціну напою.

3. Створити клас CoffeeShop з полями: 
- назва закладу
- список столиків(для перевірки наявності вільних столиків bool* m_Tables[TABLES_COUNT])
- список замовлень Coffee* m_Orders[TABLES_COUNT], де індексу вектора відповідає номер столика, який зробив замовлення

4.Реалізувати методи CoffeeShop:

int takePlace() - перевіряє наявність вільних місць, повертає номер зайнятого столика, або -1, якщо місця немає.
void order(int tableNumber) - друкує можливі варіанти напоїв, просить користувача ввести номер, який відповідає напою з меню, додає замовлення в список m_Orders.
bool prepare(int tableNumber) - готує напої для даного столика, false - якщо столик ще не зробив замовлення.
bool getReceipt(int tableNumber) - виводить вартість замовлення, звільняє столик, та видаляє замовлення, false - якщо замовлення не існує.

Протестувати роботу CoffeeShop в різних сценаріях;
```

### CODE REVIEW

**Comment by ShekhaninKyryl** [comment]
Console output: 
```
3 Clients would like to make order.
Table #0 has been taken.
Choose type of coffee:
1. Arabica
2. Black
3. Cappuccino
4. Espresso
5. Latte
6. Robusta

 Your choose: 1
Coffee is prepearing...
Type: Arabica
Cost: 1

Table #1 has been taken.
Choose type of coffee:
1. Arabica
2. Black
3. Cappuccino
4. Espresso
5. Latte
6. Robusta

 Your choose: 2
Coffee is prepearing...
Type: Black
Cost: 0.5

Sorry no more free place...
Table #1 want get receipt.
You order cost: 0.5
1 Clients would like to make order.
Table #1 has been taken.
Choose type of coffee:
1. Arabica
2. Black
3. Cappuccino
4. Espresso
5. Latte
6. Robusta

 Your choose: 3
Coffee is prepearing...
Type: Cappuccino
Cost: 0.75


E:\CPP\r_d16\x64\Debug\r_d16.exe (process 14180) exited with code 0.
```

File: r_d16/Reactangle.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+public:
+	Rectangle();
+	Rectangle(float length, float height);
+
+	float getArea();
+	float getPerimeter();
+
+	float getLength() { return length; }
+	float getHeigth() { return height; }
```
Comment (bohdanpc) [inline r_d16/Reactangle.h:14]: all of these are const methods

File: r_d16/Coffee.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class Coffee
+{
+public:
+	Coffee();
+	Coffee(CoffeeType type);
+
+	void prepare();
+
+	float getCost() { return cost; }
```
Comment (bohdanpc) [inline r_d16/Coffee.h:23]: const

File: r_d16/CoffeeShop.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+
+
```
Comment (bohdanpc) [inline r_d16/CoffeeShop.h:4]: 😄

File: r_d16/CoffeeShop.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#include <iostream>
+#include "Coffee.h"
+
+class CoffeeShop
+{
+public:
+	CoffeeShop(std::string title, unsigned int tablesCount);
```
Comment (bohdanpc) [inline r_d16/CoffeeShop.h:11]: const std::stirng&

File: r_d16/CoffeeShop.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	return true;
+}
+
+bool CoffeeShop::getReceipt(int tableNumber)
+{
+	if (!m_Tables[tableNumber]) return false;
+
+	std::cout << "You order cost: " << m_Orders[tableNumber].getCost() << std::endl;
+	m_Tables[tableNumber] = false;
+	m_Orders[tableNumber].~Coffee();
```
Comment (bohdanpc) [inline r_d16/CoffeeShop.cpp:97]: No explicit destructor should be ever called in normal condition. 
Use either `dummy empty Coffee` object or make array of Coffee pointers, with that you could just make nullptr:
```
delete m_Orders[tableNumber];
m_Orders[tableNumber] = nullptr;
```

**Pull request merged**